### PR TITLE
Setup UI test with Wallaby

### PR DESCRIPTION
**Pull request by anduonghien**
https://github.com/nimblehq/elixir-templates/issues/11

## What happened

Setup UI test with Wallaby
 
## Insight

1/ Setup Wallaby

- Running with ChromeDriver and Chrome Headless as default

- Disable Chrome Headless by export the `CHROME_HEADLESS = false`

- Take a screenshot of failing automatically and upload it to Github Action artifact

2/ Modify the CI workflow to run asset compile.
 
## Proof Of Work

I added 1 UI test to the template, the test is passed on CI now.

For testing the screenshot on failing UI test, I have trigger a failing test [here](https://github.com/nimblehq/elixir-templates/actions/runs/202085721) from there we could download the screenshot

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -67,10 +68,20 @@ defmodule Nimble.MixProject do
...
+        "ecto.migrate",
+        "assets.compile --quiet",
+        "test"
+      ],
+      "assets.compile": &compile_assets/1
     ]
   end
+
+  defp compile_assets(_) do
+    Mix.shell().cmd("cd assets && ./node_modules/.bin/webpack --mode development", quiet: true)
```
Comment (Tolsee) [inline mix.exs:85]: Let's add `build:dev` to `assets/package.json` instead?

**Comment by Tolsee** [comment]
@anduonghien Looks good to me other than a small comment. Also, rebase required  🚀

File: mix.exs (hunk 1 above)
Comment (anduonghien) [reply mix.exs]: Added in https://github.com/nimblehq/elixir-templates/pull/17/commits/8f1421b68a8e0ab15635e4742d56f9490db2217a @Tolsee

File: mix.exs [hunk 2]
```diff
@@ -67,10 +68,20 @@ defmodule Nimble.MixProject do
...
+        "ecto.migrate",
+        "assets.compile --quiet",
+        "test"
+      ],
+      "assets.compile": &compile_assets/1
     ]
   end
+
+  defp compile_assets(_) do
+    Mix.shell().cmd("cd assets && npm run build:dev", quiet: true)
```
Comment (Tolsee) [inline mix.exs:85]: 🤔 
```suggestion
    Mix.shell().cmd("npm run --prefix assets build:dev", quiet: true)
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -26,25 +26,39 @@ jobs:
...
         run: mix compile --warnings-as-errors
-        env: 
+        env:
           MIX_ENV: test
 
+      - name: Install node module
+        run: cd assets && npm install
+
+      - name: Compile assets
+        run: cd assets && ./node_modules/.bin/webpack --mode development
```
[the remark this answers is not in the document]
Comment (anduonghien) [reply .github/workflows/test.yml]: Fixed it in https://github.com/nimblehq/elixir-templates/pull/17/commits/3ab0824fdbf8ff49c304cc194377a4728cbb964d

File: mix.exs (hunk 2 above)
Comment (anduonghien) [reply mix.exs]: Fixed it in https://github.com/nimblehq/elixir-templates/pull/17/commits/3ab0824fdbf8ff49c304cc194377a4728cbb964d

File: config/test.exs [hunk 4]
```diff
@@ -12,7 +12,16 @@ config :nimble, Nimble.Repo,
...
+
+config :nimble_web, :sql_sandbox, true
 
 # Print only warnings and errors during test
 config :logger, level: :warn
+
+# Wallaby
+config :wallaby,
+  otp_app: :nimble,
+  chromedriver: [headless: System.get_env("CHROME_HEADLESS", "true") !== "false"],
```
[the remark this answers is not in the document]
Comment (anduonghien) [reply config/test.exs]: Fixed it in https://github.com/nimblehq/elixir-templates/pull/17/commits/d00c9ebabd98ee417991b22477c2fe5afc486a23 @rosle

**Pull request merged**